coding-standards/elixir.md

```diff
@@ -0,0 +1,52 @@
+# Elixir coding standards
+
+## Style Guide
+
+Trento follows the [Credo style guide](https://github.com/rrrene/elixir-style-guide).
+
+### Linting
+
+The linting is enforced by the [credo tool](https://github.com/rrrene/credo) locally and in the CI.
+
+Use the [.credo.exs template](../templates/.credo.exs) as a starting point for your project.
+
+### Additional rules and exceptions
+
+- Private functions must appear after public functions in order of usage.
+
+## Static analysis
+
+[Dialyzer](https://github.com/jeremyjh/dialyxir) must be used to check for type correctness.
+
+Please write specs `@spec` tags for all public functions and typespecs for defined types, to help Dialyzer doings its job.
```

Comment:
Should we add this to the `Additional rules and exceptions` section?

Reply:
it's not a "style" rule, so i don't think it should go there

Reply:
As you wish :wink:

I just thought that similarly to `Private functions must appear after public functions in order of usage` we could have `All public functions must have specs @spec tags` and why not `All defined types must have typespecs`

For me to better understand: since those are not "style" rules, as you say, what are they? What's the common understanding we should have?